doc/manual/documentation.rst

```diff
@@ -144,6 +144,23 @@ be written as:
         f(x) = x
     end
 
+It is possible to make use of metaprogramming to assist in the creation of documentation.
+When using string-interpolation within the docstring you will need to use an extra ``$`` as
+shown with ``$($name)``:
+
+.. code-block:: julia
+
+    for func in (:day, :dayofmonth)
+        name = string(func)
+        @eval begin
+            @doc """
+                $($name)(dt::TimeType) -> Int64
+
+            The day of month of a `Date` or `DateTime` as an `Int64`.
+            """ $func(dt::TimeType)
```

Comment:
Notice you don't need to convert symbols to strings in order to interpolate them as strings. I think something like this looks better:

```julia
for func in (:foo, :bar)
    docstring = """
        $func(dt::TimeType) -> Int64
    The day of month of a `Date` or `DateTime` as an `Int64`
    """
    @eval begin
        @doc $docstring ->
        $func(dt::TimeType)
    end
end
```

But I still think the double interpolation should be documented, this is another advantage of the proposed syntax in https://github.com/JuliaLang/julia/issues/25178.

Instead of:

```julia
for func in (:foo, :bar)
    @eval begin
        @doc """
            $($func)(dt::TimeType) -> Int64
        The day of month of a `Date` or `DateTime` as an `Int64`
        """ ->
        $func(dt::TimeType)
    end
end
```

It could be written as:

```julia
for func in (:foo, :bar)
    @eval begin
        @doc """
            \($func)(dt::TimeType) -> Int64
        The day of month of a `Date` or `DateTime` as an `Int64`
        """ ->
        $func(dt::TimeType)
    end
end
```

Making a clear distinction between string and expression interpolation, so I think that in this case:

> As a counterpoint, we use $ consistently for interpolation not just into strings, but also for command objects and expressions – i.e. of values into expressions quoted with :( ... ) and quote ... end.

This is actually an advantage.

Reply:
I have to say that in practice, I cannot for the life of me figure out the interaction between expression and string interpolation in contexts like this, so I just end up not mixing them. I'm not 100% sold on the `"\(foo)"` interpolation syntax, but having a different syntax would certainly help considerably to clarify this.